libs/langchain/langchain/document_loaders/parsers/pdf.py

```diff
@@ -149,3 +149,79 @@ def lazy_parse(self, blob: Blob) -> Iterator[Document]:
                 )
                 for page in doc.pages
             ]
+
+
+class AmazonTextractPDFParser(BaseBlobParser):
+    """Sends PDF files to Amazon Textract and parses them to generate Documents.
+    For parsing multi-page PDFs, they have to reside on S3.
+    """
+
+    try:
+        import textractcaller as tc
+    except ImportError:
+        raise ModuleNotFoundError(
+            "Could not import amazon-textract-caller python package. "
+            "Please install it with `pip install amazon-textract-caller`."
+        )
+
+    def __init__(
+        self, textract_features: List[tc.Textract_Features] = [], client: Any = None
+    ) -> None:
+        """
+        :func `Textract_features <tc.Textract_Features>`
+        """
+        self.textract_features = textract_features
+        self.boto3_textract_client = client
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Iterates over the Blob pages and returns an Iterator with a Document
+        for each page, like the other parsers If multi-page document, blob.path
+        has to be set to the S3 URI and for single page docs the blob.data is taken
+        """
+        try:
+            from urllib.parse import urlparse
+
+            import textractcaller as tc
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Could not import amazon-textract-caller python package. "
+                "Please install it with `pip install amazon-textract-caller`."
+            )
+
+        url_parse_result = urlparse(str(blob.path)) if blob.path else None
+        # Either call with S3 path (multi-page) or with bytes (single-page)
+        if (
+            url_parse_result
+            and url_parse_result.scheme == "s3"
+            and url_parse_result.netloc
+        ):
+            textract_response_json = tc.call_textract(
+                input_document=str(blob.path),
+                features=self.textract_features,
+                boto3_textract_client=self.boto3_textract_client,
+            )
+        else:
+            textract_response_json = tc.call_textract(
+                input_document=blob.as_bytes(),
+                features=self.textract_features,
+                call_mode=tc.Textract_Call_Mode.FORCE_SYNC,
+                boto3_textract_client=self.boto3_textract_client,
+            )
+
+        current_text = ""
+        current_page = 1
+        for block in textract_response_json["Blocks"]:
```

Comment:
would be helpful to document this logic -- there's concatenation going on for the last document yielded, it's unclear what the difference between the last document yielded and the ones yielded from the inner for loop. would users want some control here over what concatenation is taking place?